crates/ironrdp-cliprdr/src/pdu/format_data/mod.rs

```diff
@@ -23,6 +25,19 @@ pub struct FormatDataResponse<'a> {
     data: Cow<'a, [u8]>,
 }
 
+impl_pdu_borrowing!(FormatDataResponse<'_>, OwnedFormatDataResponse);
+
+impl IntoOwnedPdu for FormatDataResponse<'_> {
+    type Owned = OwnedFormatDataResponse;
+
+    fn into_owned_pdu(self) -> Self::Owned {
+        OwnedFormatDataResponse {
+            is_error: self.is_error,
+            data: Cow::Owned(self.data.into_owned()),
+        }
+    }
+}
```

Comment:
cc @pacmancoder 
This needs to be better documented, but that’s how we handle PDUs that may either borrows or owns data in other parts of IronRDP